### DISCUSSION TITLE
[Question] How can I scale-out in a right way?

### DISCUSSION THREAD

**Discussion by ElFisherMan** [taskforcesh/bullmq#1745] [Q&A]
I'm using BullMQ inside a service, such that it will serve all other services in the environment.
I have a concern that one instance of BullMQ will not be enouge.
What metric should I rely on in kubernetes such that the HPA will scale out, whenever there is a delay in firing sheculed tasks?
(One advice I got is using the length of the Redis bull key - wait, WDYT?)

**Comment by rosslavery** [comment]
Internally we use this package I open sourced to do something similar.

Basically you give it a URL to ping and based on the response, it will scale up or down a deployment to help process the jobs at the desired pace. Whatever number you return from the endpoint is what it would use to scale, so you could use total waiting, or total waiting plus delayed, it's up to you. 

Admittedly, the docs are very lacking but you can look at the source and see it's quite simple. 

https://github.com/rosslavery/k8s-bull-autoscaler
